### PR TITLE
Treat ansible-lint warnings as errors

### PR DESCRIPTION
**Pull request by nre-ableton**
This is necessary to prevent ansible-lint warnings from slipping through.
Unfortunately, ansible-lint doesn't plan on adding a feature to treat warnings
as errors, see https://github.com/ansible-community/ansible-lint/issues/236

### CODE REVIEW

**Comment by nre-ableton** [comment]
Here's proof of a build failure with an ansible-lint warning (the one fixed by fec0221:

https://newci.office.ableton.com/blue/organizations/jenkins/Ableton%2Fansible-role-jenkins-jcasc%2Fnre%2Fmain%2Fansible-lint-warnings/detail/ansible-lint-warnings/1/pipeline/

File: Jenkinsfile [hunk 1]
```diff
@@ -11,7 +11,21 @@ devToolsProject.run(
...
+          returnStdout: true,
+          script: 'ansible-lint -c .ansible-lint.yml',
+        )
+
+        // If only warnings are found, ansible-lint will exit with code 0 but still write
+        // an error summary to stdout. It's not possible to treat warnings as errors, and
+        // likely will never be. See:
+        // https://github.com/ansible-community/ansible-lint/issues/236
+        if (stdout) {
+          error 'ansible-lint exited with warnings'
```
Comment (mst-ableton) [inline Jenkinsfile:26]: Should we print said warnings? Or will it be in the log anyway?

**Review by mst-ableton — COMMENTED** [review]
Code looks good, I just had one question.

File: Jenkinsfile (hunk 1 above)
Comment (mst-ableton) [reply Jenkinsfile:26]: Ah nevermind, looking at the output the list is there.

Comment (mst-ableton) [reply Jenkinsfile:26]:
```suggestion
          error 'ansible-lint exited with warnings, please check the output of the previous step'
```

**Review by mst-ableton — CHANGES_REQUESTED** [review]

**Review by mst-ableton — APPROVED** [review]

**Pull request merged**